File: android/src/main/java/com/visioncameramlkit/domain/models/TextRecognitionResult.kt
package com.visioncameramlkit.domain.models

data class TextRecognitionResult(
  val text: String?,
  val blocks: List<TextBlock>,
)

data class TextBlock(
  val text: String,
  val bounds: BoundingBox?,
  val corners: List<Corner>?,
  val lines: List<TextLine>,
  val languages: List<String>,
)

data class TextLine(
  val text: String,
  val bounds: BoundingBox?,
  val corners: List<Corner>?,
  val elements: List<TextElement>,
  val confidence: Double?,
  val angle: Double?,
  val languages: List<String>,
)

data class TextElement(
  val text: String,
  val bounds: BoundingBox?,
  val corners: List<Corner>?,
  val symbols: List<TextSymbol>,
  val confidence: Double?,
  val angle: Double?,
  val languages: List<String>,
)

data class TextSymbol(
  val text: String,
  val bounds: BoundingBox?,
  val corners: List<Corner>?,
  val confidence: Double?,
  val angle: Double?,
  val languages: List<String>,
)

File: ios/Domain/Models/TextRecognitionResult.swift
import Foundation

struct TextRecognitionResult {
  let text: String?
  let blocks: [TextBlock]
}

struct TextBlock {
  let text: String
  let bounds: BoundingBox?
  let corners: [Corner]?
  let lines: [TextLine]
  let languages: [String]
}

struct TextLine {
  let text: String
  let bounds: BoundingBox?
  let corners: [Corner]?
  let elements: [TextElement]
  let confidence: Double?
  let angle: Double?
  let languages: [String]
}

struct TextElement {
  let text: String
  let bounds: BoundingBox?
  let corners: [Corner]?
  let symbols: [TextSymbol]
  let confidence: Double?
  let angle: Double?
  let languages: [String]
}

struct TextSymbol {
  let text: String
  let bounds: BoundingBox?
  let corners: [Corner]?
  let confidence: Double?
  let angle: Double?
  let languages: [String]
}
